principles/solid/liskvsustitution.md

```diff
@@ -1,22 +1,69 @@
 ## Liskov Substitution
 
-[TODO]
+It declares that a subclass must be replaceable by its superclass, and if by doing this, the program crashes, we will be violating this principle.
+
+Adhering to this principle will confirm that our program has an easy-to-understand class hierarchy and reusable code.
 
 ### Bad example:
 
-[TODO]
+```JAVA:
+    public static void printNumSeats(Car[] arrayCars){
+    for (Car car : arrayCars) {
+        if(instanceof Renault car)
+            System.out.println(numSeatsRenault(car));
+        if(instanceof Audi car)
+            System.out.println(numSeatsAudi(car));
+        if(instanceof Mercedes car)
+            System.out.println(numSeatsMercedes(car));
+    }
+    }
+    printNumSeats(arrayCars);
+```
+
+This violates both the Liskov substitution principle and the open/closed principle. The program must know each type of Car and call its associated numSeats() method. So if we add a new car, the method must be modified to accept it.
 
 ### Good example:
 
-[TODO]
+For this method to comply with the principle, we will follow these principles:
 
-### Related principles
+- If the superclass (Car) has a method that accepts a parameter of the type of the superclass (Car), then its subclass
+  (Renault) should accept a type of the superclass (Car) or a type of the subclass (Renault) as its argument.
+- If the superclass returns a type of itself (Car), then its subclass (Renault) should either return a type of the
+  superclass (Car) or a type of the subclass (Renault).
+
+```JAVA:
```

Comment:
Where is this? http://univirtual.unicauca.edu.co/moodle/mod/page/view.php?id=77008